Foundation/Unit.swift

```diff
@@ -67,62 +98,189 @@ public class Unit : NSObject, NSCopying, NSSecureCoding {
         return self
     }
     
-    public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
-    public func encode(with aCoder: NSCoder) { NSUnimplemented() }
+    public required init?(coder aDecoder: NSCoder) {
+        if aDecoder.allowsKeyedCoding {
+            guard let symbol = aDecoder.decodeObject(forKey: "symbol") as? String
+                else { return nil }
+            self.symbol = symbol
+        } else {
+            guard let symbol = aDecoder.decodeObject() as? String
+                else { return nil }
+            self.symbol = symbol
+        }
+    }
+    
+    public func encode(with aCoder: NSCoder) {
+        if aCoder.allowsKeyedCoding {
+            aCoder.encode(self.symbol.bridge(), forKey:"symbol")
+        } else {
+            aCoder.encode(self.symbol.bridge())
+        }
+    }
+
     public static func supportsSecureCoding() -> Bool { return true }
 }
 
 public class Dimension : Unit {
     
     
-    public var converter: UnitConverter { get { NSUnimplemented() } }
+    public private(set) var converter: UnitConverter
     
     
-    public init(symbol: String, converter: UnitConverter) { NSUnimplemented() }
+    public init(symbol: String, converter: UnitConverter) {
+        self.converter = converter
+        super.init(symbol: symbol)
+    }
     
     /*
      This class method returns an instance of the dimension class that represents the base unit of that dimension.
      e.g.
      NSUnitSpeed *metersPerSecond = [NSUnitSpeed baseUnit];
      */
-    public class func baseUnit() -> Self { NSUnimplemented() }
+    public class func baseUnit() -> Self { fatalError() }
```

Comment:
Let's put a message in the fatal error so people know what they did wrong. Here is the one from Darwin:

`*** You must override baseUnit in your class to define its base unit.`